tests/benchmarks/posix/threads/README.rst

```diff
@@ -0,0 +1,45 @@
+POSIX Thread Benchmark
+######################
+
+Overview
+********
+
+This benchmark creates and joins as many threads as possible within a configurable time window.
+It provides a rough comparison Zephyr's POSIX threads (pthreads) with Zephyr's kernel threads
+(k_threads) API, highlighting the overhead of the POSIX. Ideally, this overhead would shrink over
+time.
+
+Sample output of the benchmark::
+
+    *** Booting Zephyr OS build v4.0.0-1410-gfca33facee37 ***
+    ASSERT: y
+    BOARD: qemu_riscv64
+    NUM_CPUS: 1
+    TEST_DELAY_US: 0
+    TEST_DURATION_S: 5
+    SMP: n
+    API, Thread ID, time(s), threads, cores, rate (threads/s/core)
+    k_thread, ALL, 5, 47663, 1, 9532
+    pthread, ALL, 5, 28180, 1, 5636
```

Comment:
I think an efficiency metric (# of `pthread`s ÷ # of `k_thread`s x 100%) here would be nice

Reply:
Probably. The record regex is very close to the 100-character limit in yaml though, so I will probably leave that for a later improvement.